server/conn.go

```diff
@@ -1105,6 +1105,7 @@ func (cc *clientConn) Run(ctx context.Context) {
 		startTime := time.Now()
 		err = cc.dispatch(ctx, data)
 		cc.chunkAlloc.Reset()
+		cc.ctx.GetSessionVars().EndAlloc()
```

Comment:
`EndAlloc` every query? so how is the object reused?

Reply:
The alloc object in the connection-level cc.chunkAlloc, and is referenced to the session-level pool every time it needs to be used.